src/app/directives/app-role-required.directive.ts

```diff
@@ -10,8 +11,8 @@ export class RoleRequiredDirective implements OnInit {
   userRolesRequired: string[] | undefined;
 
   @Input()
-  set appRoleRequired(roles: Roles) {
-    this.userRolesRequired = roles?.split(',');
+  set appRoleRequired(roles: Roles | Roles[]) {
+    Array.isArray(roles) ? (this.userRolesRequired = uniq(roles.flatMap(role => role.split(',')))) : (this.userRolesRequired = roles?.split(','));
```

Comment:
Don't know if I got the brackets right but:
```suggestion
    Array.isArray(roles) ? (this.userRolesRequired = [...new Set(roles.flatMap(role => role.split(','))))] : (this.userRolesRequired = roles?.split(','));
```

Reply:
Thank you, I was wondering how I was going to get it into a set